taichi/codegen/codegen_llvm.cpp

```diff
@@ -1571,6 +1571,13 @@ void CodeGenLLVM::create_offload_struct_for(OffloadedStmt *stmt, bool spmd) {
 
   llvm::Function *body = nullptr;
   auto leaf_block = stmt->snode;
+  // generate struct for on bit array's parent node(usually dense) instead of
+  // itself
+  if (leaf_block->type == SNodeType::bit_array && leaf_block->parent &&
+      leaf_block->parent->type == SNodeType::dense) {
+    leaf_block = leaf_block->parent;
+  }
```

Comment:
Maybe add a loud error message here (`TI_ERROR`) if we are looping over a bit array but its parent is not dense?